src/lib/field.ts

```diff
@@ -376,6 +376,63 @@ class Field {
     return this.add(Field.from(y).neg());
   }
 
+  #parity() {
+    if (this.isConstant()) return new Field(this.toBigInt() % 2n);
+
+    let [, isOddVar, xDiv2Var] = Snarky.exists(2, () => {
+      let bits = Fp.toBits(this.toBigInt());
+      let isOdd = bits.shift()! ? 1n : 0n;
```

Comment:
Why do you need the `!` on `bits.shift()`, that method doesn't return undefined, as far as I understand

Reply:
it returns `undefined` if the array is empty, but since we can be sure `bits` will never be empty, we can just "ignore" it like that